src/components/KMultiselect/KMultiselect.cy.ts

```diff
@@ -576,4 +576,23 @@ describe('KMultiselect', () => {
     cy.get('.k-multiselect-item').eq(3).should('contain.text', items[2].label)
     cy.get('.k-multiselect-item').eq(4).should('contain.text', items[4].label)
   })
+
+  it('should able to handle tons of items with no obvious lag', () => {
+    const items = Array.from(new Array(500)).map((_, i) => ({
+      label: `Item ${i}`,
+      value: `${i}`,
+      selected: i < 400,
+    }))
+
+    const startTime = Date.now()
+
+    mount(KMultiselect, {
+      props: {
+        testMode: true,
+        items,
+      },
+    }).then(() => {
+      expect(Date.now() - startTime).to.be.lessThan(3000)
```

Comment:
question: why did you choose 3 seconds? Isn’t that still a considerable amount of lag?

Reply:
Rendering 500 items does generate some pressure on the browser. On my Macbook, it took me ~380ms to render the component. Considering the CI machine is not that powerful, multiple 7-8 looks safe to me, so I chose 3 seconds.
Current implementation on CI takes ~1.2s, in comparison, the original implementation will take 40+ seconds or even cause the page to crash. I have to say the 3 seconds is a bit arbitrary, but I think it's okay to be the bottom line

Reply:
As long as we don't consistently see failures in the CI, that makes sense. Thanks for the extra info